File: cakk-common/src/main/java/com/cakk/common/enums/VerificationStatus.java
package com.cakk.common.enums;

public enum VerificationStatus {

	UNREQUESTED(0),
	APPROVED(1),
	REJECTED(2),
	PENDING(3);

	private final Integer code;

	VerificationStatus(Integer code) {
		this.code = code;
	}

	public Integer getCode() {
		return code;
	}

	public boolean isCandidate() {
		return code == 3;
	}

	public boolean isNotCandidate() {
		return  code != 3;
	}

	public boolean isApproved() {
		return code == 1;
	}

	public boolean isRejected() {
		return code == 2;
	}

	public VerificationStatus makeApproved() {
		return VerificationStatus.APPROVED;
	}

	public VerificationStatus makePending() {
		return VerificationStatus.PENDING;
	}
}

File: cakk-common/src/main/kotlin/com/cakk/common/enums/VerificationStatus.kt
package com.cakk.common.enums

enum class VerificationStatus(@JvmField val code: Int) {
	UNREQUESTED(0),
	APPROVED(1),
	REJECTED(2),
	PENDING(3);

	val isCandidate: Boolean
		get() = code == 3

	val isNotCandidate: Boolean
		get() = code != 3

	val isApproved: Boolean
		get() = code == 1

	val isRejected: Boolean
		get() = code == 2

	fun makeApproved(): VerificationStatus {
		return APPROVED
	}

	fun makePending(): VerificationStatus {
		return PENDING
	}
}

